TheSadRogue.Primitives.UnitTests/RectangleTests.cs

```diff
@@ -105,5 +121,139 @@ public void TestTupleEquality(Rectangle rect)
+            Assert.True(rectangles[1].Height <= 11);
+            Assert.Equal(5, rectangles[0].Width);
+            Assert.Equal(5, rectangles[1].Width);
+
+            rectangle = new Rectangle(0, 0, 13, 5);
+            rectangles.AddRange(rectangle.Bisect().ToList());
+            foreach (Point c in rectangles[2].Positions())
+            {
+                Assert.True(rectangle.Contains(c));
+                Assert.False(rectangles[3].Contains(c));
+            }
+
+            foreach (Point c in rectangles[3].Positions())
+            {
+                Assert.True(rectangle.Contains(c));
+                Assert.False(rectangles[2].Contains(c));
+            }
+
+            Assert.True(rectangles[2].Width >= 3);
+            Assert.True(rectangles[2].Width <= 10);
+            Assert.True(rectangles[3].Width >= 3);
+            Assert.True(rectangles[3].Width <= 10);
+            Assert.Equal(5, rectangles[2].Height);
+            Assert.Equal(5, rectangles[3].Height);
+        }
+
+        [Fact]
+        public void BisectHorizontallyTest()
+        {
+            Rectangle rectangle = new Rectangle(0, 0, 5, 13);
+            List<Rectangle> rectangles = rectangle.BisectHorizontally().ToList();
+            foreach (Point c in rectangles[0].Positions())
+            {
+                Assert.True(rectangle.Contains(c));
+                Assert.False(rectangles[1].Contains(c));
+            }
+
+            Assert.True(rectangles[0].Height >= 3);
+            Assert.True(rectangles[0].Height <= 10);
+            Assert.True(rectangles[1].Height >= 3);
+            Assert.True(rectangles[1].Height <= 10);
+            Assert.Equal(5, rectangles[0].Width);
+            Assert.Equal(5, rectangles[1].Width);
+
+        }
+
+        [Fact]
+        public void BisectVerticallyTest()
+        {
+            Rectangle rectangle = new Rectangle(0, 0, 13, 5);
+            List<Rectangle> rectangles = rectangle.BisectVertically().ToList();
+            foreach (Point c in rectangles[0].Positions())
+            {
+                Assert.True(rectangle.Contains(c));
+                Assert.False(rectangles[1].Contains(c));
+            }
+
+            Assert.True(rectangles[0].Width >= 3);
+            Assert.True(rectangles[0].Width <= 10);
+            Assert.True(rectangles[1].Width >= 3);
+            Assert.True(rectangles[1].Width <= 10);
+            Assert.Equal(5, rectangles[0].Height);
+            Assert.Equal(5, rectangles[1].Height);
+        }
+
+        #endregion
+
+        #region Division
+        [Theory]
+        [MemberDataTuple(nameof(DivisionTestData))]
+        public void DivisionOperatorTest(Rectangle dividend, Rectangle divisor)
+        {
+            int expectedColumns = dividend.Width / divisor.Width;
+            int expectedRows = dividend.Height / divisor.Height;
+            int expectedRectangles = expectedRows * expectedColumns;
+
+            var answer = dividend / divisor;
```

Comment:
This is a good example of why an operator specifically, rather than a function, returning an iterable-generated `IEnumerable` is architecturally questionable for me.  I didn't think about it previously, but looking at this the issue is this lends itself to creating "multiple enumeration" issues.  For example, 2 for-loops iterating over `answer` now causes the entire calculation to be done twice.

How we normally get around this type of thing in functions, is to just perform the function call in-line with the for-loop; ex.
```CSharp
foreach (var rect in rect1.Divide(rect2))
   ...
```

My issue with an operator returning this type of thing is, the following code, unlike the previous, does _not_ seem intuitive (in fact I'm not sure it's legal in C#)
```CSharp
foreach (var rect in rect1 / rect2)
    ....
```

If you disagree that this is an issue, feel free to say so and we can discuss; however assuming that we concur this is an issue I can think of 2 solutions:
1. Change the return type of `operator/` to `List<Rectangle>` so this won't happen.  This solves this issue but does make the code potentially less performant (by virtue of creating temporary lists).
2. Simply remove the `operator/` and leave only the `Divide` function.  This is somewhat unusual; and has some implications for function naming.  That much said, it doesn't directly defy the C#/.NET guidelines (I believe they say if you have an `operator/`, you should have a corresponding `Divide` function; but not the other way around explicitly).

Off the top of my head, personally I think option 2 might be the best for encouraging clean code, but open to suggestions/discussion.

Reply:
You know, I think I agree. I've removed the operator and have left it with just the function.